tests/run/switches.scala

```diff
@@ -18,7 +18,7 @@ object Test extends App {
     case _ => 4
   }
 
-  val x3 = (x: @switch) match {
+  val x3 = x match {
     case '0' if x > 0 => 0
```

Comment:
scalac supports emitting switch when cases contain conditions, it translates something like:
```scala
  def foo(x: Int, cond: Boolean) = x match {
    case 1 if cond =>
      println("1")
    case 2 =>
      println("2")
    case 3 =>
      println("3")
    case _ =>
      println("4")
  }
```
to
```scala
      x1 match {
        case 1 => if (cond)
          scala.Predef.println("1")
        else
          default5()
        case 2 => scala.Predef.println("2")
        case 3 => scala.Predef.println("3")
        case _ => default5(){
          scala.Predef.println("4")
        }
      }
```
Is this an intended limitation of Dotty `@switch`, or something that still needs to be implemented?

Reply:
This was supposed to go in the local optimizations. I made an implementation of this a while ago but I do not know if it went into Dotty or just stayed in the linker.

Reply:
I think it would make sense to have it by default. Could you open an issue so we keep track of it?

Reply:
> Could you open an issue so we keep track of it?

Turns out there's already one: https://github.com/lampepfl/dotty/issues/1313